orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -284,80 +293,57 @@ def forward(ctx, *inputs, **kwargs):
-                                   self._device)
+                # Push user output grads to ONNX backend.
+                backward_grad_output_ortvalue = []
+                for grad_output in grad_output[:len(self._onnx_graphs_info.backward_output_grad_names)]:
+                    backward_grad_output_ortvalue.append(onnxruntime.OrtValue.ortvalue_from_data_ptr(list(grad_output.size()), _utils.dtype_torch_to_numpy(
+                        grad_output.dtype), grad_output.device.type, _get_device_index(grad_output.device), grad_output.data_ptr()))
 
-                # Run
-                self._backward_session.run_with_iobinding(self._backward_io_binding)
-                backward_outputs = self._backward_io_binding.get_outputs()
+                # Run and get results
+                self._gradient_session.run_backward(backward_grad_output_ortvalue)
+                backward_outputs = self._gradient_io_binding.get_outputs()
 
                 # Return input and initializer gradients
-                num_initializers = len(self._onnx_graphs_info.initializer_grad_names_to_train)
+                num_user_input_grads = len(self._input_names_require_grad)
+
                 results = []
                 for input_name in self._onnx_graphs_info.user_input_names:
                     try:
                         # Append to the results the backward output for each input that required grad
                         results.append(_ort_output_to_torch_tensor(
-                            backward_outputs[num_initializers + self._input_names_require_grad.index(input_name)]))
+                            backward_outputs[self._input_names_require_grad.index(input_name)]))
                     except ValueError:
+                        # input_name is not found in the self._input_names_require_grad list
                         # Append None to results for each input that did not require grad
                         results.append(None)
-                # Append backward ouput for all trained initializers
-                results += [_ort_output_to_torch_tensor(backward_output)
-                            for backward_output in backward_outputs[:num_initializers]]
+                # Append gradients of initializer to results
+                results += [_ort_output_to_torch_tensor(backward_output) 
+                            for backward_output in backward_outputs[num_user_input_grads:]]
                 return tuple(results)
 
         proc_inputs = [data for data in inputs if data is not None]
-        return _ORTModuleFunction.apply(*self._convert_forward_input_to_list(*proc_inputs, **kwargs))
+
+        return _ORTModuleFunction.apply(*self._convert_gradient_graph_input_to_list(*proc_inputs, **kwargs))
 
     @_utils.timeit(enabled=__TEMP_ENABLE_METHOD_TIMING__)
-    def _convert_forward_input_to_list(self, *inputs, **kwargs):
+    def _convert_gradient_graph_input_to_list(self, *inputs, **kwargs):
```

Comment:
What is the motivation behind changing this name? My understanding is that ```forward``` in this context refers to the ```_ORTModuleFunction(torch.autograd.Function)``` ```forward``` method.

Reply:
There are two onnx model/graph in ORTModule: 
self._onnx_training is the "forward-only" graph
self._onnx_gradient is the "forward+backward" graph (I think we should also rename "_onnx_gradient" to something else to reflect the forward+backward reality) 

The renaming is for distinguishing the difference between "forward" and "gradient" graph.